File: src/modules/mail/domain/interfaces/mail_option_interface.py
from typing import TypedDict, TypeVar
from src.modules.mail.domain.enums.mail_template_enum import EmailTemplate
from src.modules.mail.domain.interfaces.attachment_interface import IAttachment

T = TypeVar("T")


class IMailOption(TypedDict):
    to: str | list[str]
    subject: str
    template: EmailTemplate
    context: T
    attachments: list[IAttachment] | None

File: src/modules/mail/domain/interfaces/mail-option.interface.ts
import { EMailTemplate } from '../enums/mail-template.enum';
import { IAttachment } from './attachment.interface';

export interface IMailOption<T> {
  to: string | string[];
  subject: string;
  template: EMailTemplate;
  context: T;
  attachments?: IAttachment[];
  html?: string;
}
